Samples/2.0/ChatRoom/OrleansClient/Program.cs

```diff
@@ -0,0 +1,200 @@
+		    PrettyConsole.Line("Type '/h' to re-read channel history", menuColor);
+		    PrettyConsole.Line("Type '/m' to query members in the channel", menuColor);
+            PrettyConsole.Line("Type '/exit' to exit client.", menuColor);
+		}
+
+		private static async Task Menu(IClusterClient client)
+		{
+			string input;
+			PrintHints();
+
+			do
+			{
+				input = Console.ReadLine();
+
+				if (string.IsNullOrWhiteSpace(input)) continue;
+
+				if (input.StartsWith("/j"))
+				{
+					await JoinChannel(client, input.Replace("/j", "").Trim());
+				}
+                else if (input.StartsWith("/n"))
+				{
+				    userName = input.Replace("/n", "").Trim();
+				    PrettyConsole.Line($"Your user name is set to be {userName}", ConsoleColor.DarkGreen);
+                }
+				else if (input.StartsWith("/l"))
+				{
+					await LeaveChannel(client);
+				}
+				else if (input.StartsWith("/h"))
+				{
+				    await ShowCurrentChannelHistory(client);
+				}
+				else if (input.StartsWith("/m"))
+				{
+				    await ShowChannelMembers(client);
+				}
+                else if (!input.StartsWith("/exit"))
+				{
+					await SendMessage(client, input);
+				}
+			} while (input != "/exit");
+		}
+
+	    private static async Task ShowChannelMembers(IClusterClient client)
+	    {
+	        var room = client.GetGrain<IChannel>(joinedChannel);
+	        var members = await room.GetMembers();
+
+	        PrettyConsole.Line($"====== Members for '{joinedChannel}' Channel ======", ConsoleColor.DarkGreen);
+	        foreach (var member in members)
+	        {
+	            PrettyConsole.Line(member, ConsoleColor.DarkGreen);
+	        }
+	        PrettyConsole.Line("============", ConsoleColor.DarkGreen);
+	    }
+
+        private static async Task ShowCurrentChannelHistory(IClusterClient client)
+	    {
+	        var room = client.GetGrain<IChannel>(joinedChannel);
+	        var history = await room.ReadHistory(1000);
+
+	        PrettyConsole.Line($"====== History for '{joinedChannel}' Channel ======", ConsoleColor.DarkGreen);
+	        foreach (var chatMsg in history)
+	        {
+	            PrettyConsole.Line($" ({chatMsg.Created:g}) {chatMsg.Author}> {chatMsg.Text}", ConsoleColor.DarkGreen);
+	        }
+	        PrettyConsole.Line("============", ConsoleColor.DarkGreen);
+	    }
+
+        private static async Task SendMessage(IClusterClient client, string messageText)
+		{
+			var room = client.GetGrain<IChannel>(joinedChannel);
+			await room.Message(new ChatMsg(userName, messageText));
+		}
+
+		private static async Task JoinChannel(IClusterClient client, string channelName)
+		{
+		    PrettyConsole.Line($"Joining to channel {channelName}");
+            joinedChannel = channelName;
+			var room = client.GetGrain<IChannel>(joinedChannel);
+		    var streamId = await room.Join(userName);
+            //empty stream id meaning client already joined this channel, so return 
+		    if (streamId == Guid.Empty) return;
```

Comment:
If we return the actual stream GUID, we won't have to exit here.

Reply:
yes sorry I forgtot to remove it